### PR TITLE
Added initial support for SDK 1.1.

### PR DESCRIPTION
**Pull request by slaff**
See issue: #86.

### CODE REVIEW

**Comment by anakod** [comment]
> PERIPHS_IO_MUX_PULLUP2
We just remove the PULLDOWN definition because our pins do not support inside-pulldown function now.
Functionally, It's the same as PERIPHS_IO_MUX_PULLUP.

I think we should remove all references to pull-down in Sming because previous API from Espressif was wrong, and pulldown doesn't suported

**Comment by kzyapkov** [comment]
For me, building any of the examples with SDK 1.1 results in a stream of:

    Fatal exception (0): 
    epc1=0x40105ad3, epc2=0x00000000, epc3=0x00000000, excvaddr=0x00000000, depc=0x00000000

Everything works fine with 1.0.1.

Thanks for the great framework!

**Comment by slaff** [comment]
@anakod Unfortunately I get the same result as @kzyapkov :(. Can it be that the ld script needs updating Similar to https://github.com/esp8266/Arduino/commit/92147999b997e7e167ab0365cc98d1f2e99824c7 ? This is just a blind guess.

**Comment by dmarkey** [comment]
Seems probably, anyone know how to generate a new one?

**Comment by dmarkey** [comment]
I applied it by hand, seems to work for me:

https://github.com/dmarkey/Sming/commit/8e74effdd397d9c3677cab240306ea4eeac063f2

**Comment by slaff** [comment]
@dmarkey I will check tonight on my Olimex ESP. It would be great if it works :)

**Comment by dmarkey** [comment]
@kzyapkov if you could have a go that would be helpful thanks!

**Comment by kzyapkov** [comment]
@dmarkey , yes, updating the ld script fixed it, tested on ESP-12 (nodemcu devkit) and an Olimex' module.

**Comment by dmarkey** [comment]
Hmm is there some issue with this request?

File: Sming/SmingCore/Platform/System.cpp [hunk 1]
```diff
@@ -51,7 +51,7 @@ void SystemClass::setCpuFrequency(CpuFrequency freq)
 	else
 		REG_CLR_BIT(0x3ff00014, BIT(0));
 
-	os_update_cpu_frequency(freq);
+	ets_update_cpu_frequency(freq);
```
Comment (anakod) [inline Sming/SmingCore/Platform/System.cpp]: Can you discover, why this change needed?

File: Sming/compiler/ld/eagle.app.v6.cpp.ld [hunk 2]
```diff
@@ -20,6 +20,11 @@ PHDRS
 
 /*  Default entry point:  */
 ENTRY(call_user_start)
+EXTERN(_DebugExceptionVector)
+EXTERN(_DoubleExceptionVector)
+EXTERN(_KernelExceptionVector)
+EXTERN(_NMIExceptionVector)
+EXTERN(_UserExceptionVector)
```
Comment (anakod) [inline Sming/compiler/ld/eagle.app.v6.cpp.ld]: Already merged, yes?

File: Sming/SmingCore/SdkCompat.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+#ifndef os_intr_lock
+#define os_intr_lock ets_intr_lock
+#endif
+
+#ifndef os_intr_unlock
+#define os_intr_unlock ets_intr_unlock
+#endif
+
+#endif /* SDKCOMPAT_H_ */
```
Comment (anakod) [inline Sming/SmingCore/SdkCompat.h]: I think this file should be removed, because pull-down isn't accessible in hardware (is it right, yes?). So we just should remove all relations to pull-down mode in code.
.

File: Sming/SmingCore/Platform/System.cpp (hunk 1 above)
Comment (slaff) [inline Sming/SmingCore/Platform/System.cpp]: It seems that this function, along with others, was removed from osapi.h. An abbreviated version of the diff is given below. Lines starting with "-" mean that the code was removed and lines starting with "+" mean that this code was added. I guess we have to ask Espressif for the reasons behind these changes.

```diff
diff -u /opt/Espressif/esp_iot_sdk_v1.0.1/include/osapi.h /opt/Espressif/esp_iot_sdk_v1.1.0/include/osapi.h
--- /opt/Espressif/esp_iot_sdk_v1.0.1/include/osapi.h	2015-04-21 14:45:08.000000000 +0000
+++ /opt/Espressif/esp_iot_sdk_v1.1.0/include/osapi.h	2015-05-22 19:45:30.000000000 +0000
-#define os_install_putc2 ets_install_putc2
-#define os_intr_lock ets_intr_lock
-#define os_intr_unlock ets_intr_unlock
-#define os_isr_attach ets_isr_attach
-#define os_isr_mask ets_isr_mask
-#define os_isr_unmask ets_isr_unmask

-#define os_putc ets_putc
-#define os_str2macaddr ets_str2macaddr
-#define os_timer_done ets_timer_done
-#define os_timer_handler_isr ets_timer_handler_isr
-#define os_timer_init ets_timer_init
-#define os_update_cpu_frequency ets_update_cpu_frequency
+unsigned long os_random(void);
+int os_get_random(unsigned char *buf, size_t len);
```

File: Sming/compiler/ld/eagle.app.v6.cpp.ld (hunk 2 above)
Comment (slaff) [inline Sming/compiler/ld/eagle.app.v6.cpp.ld]: Yes, that's correct.

File: Sming/SmingCore/SdkCompat.h (hunk 3 above)
Comment (slaff) [inline Sming/SmingCore/SdkCompat.h]: > I think this file should be removed, because pull-down isn't accessible in hardware (is it right, yes?). 

According to Espressif the hardware does not support this. I would say that we should listen to them and do as you suggested - remove all occurences of PULL_DOWN. 

As for the os_intr_unclock and os_intra_lock if you want I can replace them directly in the framework. 

And finally I can remove them SdkCompat.h because it won't be needed any longer.

What do you think? Comments from your side?

Comment (anakod) [inline Sming/SmingCore/SdkCompat.h]: Solution:
1. Replace all what needed
2. Remove all what doesn't needed :)

**Comment by slaff** [comment]
@anakod I rebased the changes. Please, take a look at them and check if they are ready for merging.

**Comment by anakod** [comment]
Very big thanks for all! It working on previous SDK version without any problems too.
